pkg/apis/networking/types.go

```diff
@@ -699,3 +700,58 @@ type ClusterCIDRList struct {
 	// items is the list of ClusterCIDRs.
 	Items []ClusterCIDR
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPAddress represents an IP used in a Kubernetes cluster, commonly associated to a Service ClusterIP.
```

Comment:
yeah, excellent point, the historical context is that this was going to be only a Service IPAddress, it represents a Virtual IP that has some Pod behind ... but then Gateway and GAMMA expressed their interest, since it solves some of their problems of relying on Services ... sig-network can also benefit because it allows to "evolve" Services to be able express more complex networking setups without having to keep overloading the Service object and using Gateway API ...

 I will define it as `IPAddress represents a Virtual IP in the cluster that acts as a frontend for logical a set of Pods, used for representing the Service ClusterIP, but may be used for other APIs, like Gateway`

cc: @thockin @robscott